### ISSUE TITLE
GetThreadID doesn't work reliably under new OS

### ISSUE THREAD

**Issue by zephray** [nbzwt/SDL-1.2-N3DS#1]
Sometimes it returns 0 in a sub-thread.

**Comment by zephray** [comment]
GetThreadID works as following:
```
Thread current = threadGetCurrent();
Handle handle = threadGetHandle(current);
u32 threadID;
svcGetThreadId(&threadID, handle);
```
It seems like ```threadGetHandle(current)``` sometimes return -1.

**Comment by zephray** [comment]
Fixed in #3bdd29abb65aabfead3f6a4f80ec1e7695323e6c
